src/affinity.cc

```diff
@@ -0,0 +1,83 @@
+// non-platform-specific code for thread pinning
+
+#include "affinity.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <libenvpp/env.hpp>
+
+namespace celerity::detail::thread_pinning {
+
+std::string thread_type_to_string(const thread_type t_type) {
+	switch(t_type) {
+	case thread_type::application: return "application";
+	case thread_type::scheduler: return "scheduler";
+	case thread_type::executor: return "executor";
+	default: break;
+	}
+	if(t_type >= thread_type::first_backend_worker && t_type < thread_type::first_host_queue) {
+		return fmt::format("backend_worker_{}", t_type - thread_type::first_backend_worker);
```

Comment:
I shortened this to "submit" yesterday because I somehow thought this was about naming threads (as in, setting their name using the POSIX API, where we have a 16 character limit). While these are different names, unfortunately the actual thread names are too long as well since the "worker -> submission" renaming in #303:
![image](https://github.com/user-attachments/assets/d666ffb3-6a7c-4b31-afad-f1b8ed7d92c2)

Reply:
Come to think of it, it's kind of confusing that we now have two different concepts of thread names, between what the pinner prints, and what we set using the POSIX API (with the "cy-" prefix).